src/main/java/marquez/service/mappers/DatasourceRowMapper.java

```diff
@@ -18,24 +18,32 @@
 import lombok.NonNull;
 import marquez.common.models.ConnectionUrl;
 import marquez.common.models.DatasourceName;
+import marquez.common.models.DatasourceUrn;
 import marquez.db.models.DatasourceRow;
 import marquez.service.models.DbTableVersion;
 
 public final class DatasourceRowMapper {
   private DatasourceRowMapper() {}
 
   public static DatasourceRow map(@NonNull DbTableVersion dbTableVersion) {
+    final DatasourceName datasourceName =
+        DatasourceName.fromString(dbTableVersion.getConnectionUrl().getDatasourceType().name());
+    final DatasourceUrn datasourceUrn =
+        DatasourceUrn.from(dbTableVersion.getConnectionUrl(), datasourceName);
     return DatasourceRow.builder()
         .uuid(UUID.randomUUID())
-        .name(dbTableVersion.getConnectionUrl().getDatasource().getValue())
+        .urn(datasourceUrn.getValue())
+        .name(dbTableVersion.getConnectionUrl().getDatasourceType().name())
```

Comment:
Isn't the `urn` from a datasource different than it's `name`?

Reply:
I think what happened is that in the past, the `datasource` field was used both as the type as well as naming. I originally did not write the above method and I tried to have it deviate as little as possible, so this was necessary to keep the original intent as close as possible (since tests or other behavior may have been relying on it).

As part of my next checkin, I've modified it to return the name and adjusted a test to reflect that.